### PR TITLE
Origin/net 1393 add family to pubsub update messages

### PR DESCRIPTION
**Pull request by Jovonni**
## Description

[Draft]
Refactoring to add 'model' to the UPDATE pubsub messages

Include relevant motivation, context, brief description and impact of the change(s). List follow-up tasks here.

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

- [ ] Test A (e.g. Test A - New test that ... ran in local, docker, and dev unstable.)
- [ ] Test B

## PR checklist

Before submitting this PR, please make sure:

- [ ] I have tagged the relevant reviewers and interested parties
- [ ] I have updated the READMEs of affected packages
- [ ] I have made corresponding changes to the documentation

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.

### CODE REVIEW

File: packages/core/src/__tests__/dispatcher-mock-ipfs.test.ts
Comment (stbrody) [inline packages/core/src/__tests__/dispatcher-mock-ipfs.test.ts:238]: the update function is already mocked on line 234, doing it again here is redundant.

File: packages/core/src/dispatcher.ts
Comment (stbrody) [inline packages/core/src/dispatcher.ts:314]: ?

File: packages/core/src/dispatcher.ts
Comment (stbrody) [inline packages/core/src/dispatcher.ts:316]: don't forget to clean up extra logs used during development before sending code out for review

File: packages/core/src/pubsub/pubsub-message.ts
Comment (stbrody) [inline packages/core/src/pubsub/pubsub-message.ts:24]: model should be a StreamID

File: packages/core/src/pubsub/pubsub-message.ts
Comment (stbrody) [inline packages/core/src/pubsub/pubsub-message.ts:101]: if your editor is putting these comments in please reconfigure it so it stops doing that.  Or at least make sure to take them out before putting code up for review

File: packages/core/src/pubsub/pubsub-message.ts
Comment (stbrody) [inline packages/core/src/pubsub/pubsub-message.ts:104]: what happens if model is missing?  Then the `payload` object will have `model: undefined`.  How will `textEncoder.encode()` handle that?  Will it strip out the `model` field (desired behavior), or will it still include the `model` field as an explicit `undefined` (not desired behavior as it increases message size unnecessarily)?

Comment (stbrody) [reply packages/core/src/pubsub/pubsub-message.ts:104]: instead of trying to figure out what `textEncoder` does here, it may be better to just proactively only add the `model` field to the payload if you know it's non-null

File: packages/core/src/pubsub/pubsub-message.ts
Comment (stbrody) [inline packages/core/src/pubsub/pubsub-message.ts:137]: you'll need to parse the `model` into a StreamID like is done for the `stream` field above

File: packages/core/src/state-management/state-manager.ts
Comment (stbrody) [inline packages/core/src/state-management/state-manager.ts:205]: There's no need for this function to take a `model` argument.  This function is for applying the new update to the StreamState for the updated stream - the model property doesn't affect that behavior at all.  For now, nodes that receive UPDATE messages can just ignore the model field entirely.  It'll be used later by the indexer but that isn't hooked up yet.

**Review by stbrody — CHANGES_REQUESTED** [review]
great start, some comments throughout.  There will of course need to be a follow-up commit to actually include the model when sending the update messages, but that can happen after this goes in.

File: packages/core/src/state-management/state-manager.ts
Comment (oed) [inline packages/core/src/state-management/state-manager.ts:196]: Something like this, naming is likely off.
```suggestion
    // NOTE: pass model to publishTip
    this.dispatcher.publishTip(state$.id, state$.tip, state$.state.metadata.model)
```

File: packages/core/src/pubsub/pubsub-message.ts
Comment (oed) [inline packages/core/src/pubsub/pubsub-message.ts:101]:
```suggestion
      }
      if (message.model) payload.model = message.model.toString()
```
I think we can make this a little easier to read.

File: packages/core/src/pubsub/pubsub-message.ts
Comment (oed) [inline packages/core/src/pubsub/pubsub-message.ts:129]: Does `StreamID.fromString` return undefined if `parsed.model` is not defined?

File: packages/core/src/state-management/state-manager.ts
Comment (oed) [inline packages/core/src/state-management/state-manager.ts:197]: I guess you are still working this out?

File: packages/core/src/state-management/state-manager.ts
Comment (oed) [inline packages/core/src/state-management/state-manager.ts:195]:
```suggestion
```
This comment is no longer relevant.

File: packages/core/src/pubsub/pubsub-message.ts
Comment (oed) [reply packages/core/src/pubsub/pubsub-message.ts:129]: Still curious about this.

**Review by oed — APPROVED** [review]
lgtm, one Q and one small update suggested.

**Comment by oed** [comment]
I guess also make sure tests pass :)

**Review by oed — APPROVED** [review]
I think this can be merged when the tests pass

File: packages/common/src/stream.ts
Comment (stbrody) [inline packages/common/src/stream.ts:94]: this already exists in the `develop` branch, please merge in develop branch and tag me to take another look after you do and the tests are passing

File: packages/core/src/__tests__/dispatcher-mock-ipfs.test.ts
Comment (stbrody) [inline packages/core/src/__tests__/dispatcher-mock-ipfs.test.ts:289]:
```suggestion
    // Handle UPDATE message with model
```

File: packages/core/src/__tests__/dispatcher-mock-ipfs.test.ts
Comment (stbrody) [inline packages/core/src/__tests__/dispatcher-mock-ipfs.test.ts:281]: no need for testing QUERY or RESPONSE messages in this test, they are already tested in the previous test you copied this one from, that would just be redundant duplicate testing.  I think this test case can be made way smaller and only focused on the UPDATE message

File: packages/core/src/dispatcher.ts
Comment (stbrody) [inline packages/core/src/dispatcher.ts:312]:
```suggestion
    // TODO(NET-1527) model isn't used in update yet, will be in later versions
```

**Review by stbrody — COMMENTED** [review]
basically looks good.  A few small comments, and needs to be merged with develop and the tests made to pass

**Review by ukstv — APPROVED** [review]
LGTM mod @stbrody 's comments.

**Comment by oed** [comment]
@smrz2001 Why does the new ci pipeline fail? Can it be ignored for now or should we be concerned?

**Comment by smrz2001** [comment]
> @smrz2001 Why does the new ci pipeline fail? Can it be ignored for now or should we be concerned?

@oed, it fails because of known flaky unit tests. @stbrody mentioned adding a ticket this coming cycle to clean such tests up.

My recommendation would be to ignore these till the end of next cycle, after which we can both deprecate CircleCI and enforce that the new pipeline must pass before a PR can be merged.

File: packages/core/src/pubsub/pubsub-message.ts
Comment (stbrody) [reply packages/core/src/pubsub/pubsub-message.ts:105]: looks like this is back to always adding `model` instead of only adding it if it's not undefined

File: packages/core/src/pubsub/pubsub-message.ts
Comment (stbrody) [reply packages/core/src/pubsub/pubsub-message.ts:105]: then you could also delete line 107 and keep all the construction of this object together in one place

File: packages/core/src/pubsub/pubsub-message.ts
Comment (oed) [reply packages/core/src/pubsub/pubsub-message.ts:105]: I don't think this matters since `JSON.stringify` (on line 108) removes all properties that are not defined.

Comment (oed) [reply packages/core/src/pubsub/pubsub-message.ts:105]: `undefined` is not a thing in JSON.

Comment (stbrody) [reply packages/core/src/pubsub/pubsub-message.ts:105]: Okay cool, in that case you can probably just delete line 107 and call it a day

**Pull request merged**
